### PR TITLE
fix webhooks and try to fix e2e

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
1. Add default for rolloutStrategy
2. Attempt to fix flaky e2e validation

### CODE REVIEW

File: apis/standard.oam.dev/v1alpha1/rollout_state.go [hunk 1]
```diff
@@ -265,9 +265,12 @@ func (r *RolloutStatus) StateTransition(event RolloutEvent) {
 	}
 	// special handle modified event here
 	if event == RollingModifiedEvent {
+		if r.RollingState == RolloutDeletingState {
+			panic(fmt.Errorf(invalidRollingStateTransition, rollingState, event))
```
Comment (wonderflow) [inline apis/standard.oam.dev/v1alpha1/rollout_state.go:269]: Does it necessary to panic? Can we set some state that represent this `panic`? My concern is the approllout panic will affect the whole kubevela controller

File: pkg/controller/common/rollout/workloads/cloneset_scale_controller.go [hunk 2]
```diff
@@ -272,12 +272,20 @@ func (s *CloneSetScaleController) Finalize(ctx context.Context, succeed bool) bo
...
 	for _, owner := range s.cloneSet.GetOwnerReferences() {
 		if owner.Kind == v1beta1.AppRolloutKind && owner.APIVersion == v1beta1.SchemeGroupVersion.String() {
+			isOwner = true
 			continue
 		}
 		newOwnerList = append(newOwnerList, owner)
 	}
+	if !isOwner {
+		// nothing to do if we are already not the owner
+		klog.InfoS("the cloneset is already released", "cloneSet", s.cloneSet.Name)
```
Comment (wonderflow) [inline pkg/controller/common/rollout/workloads/cloneset_scale_controller.go:285]:
```suggestion
		klog.InfoS("the cloneset is already released and not controlled by rollout", "cloneSet", s.cloneSet.Name)
```

File: pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go [hunk 3]
```diff
@@ -111,10 +112,12 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (res reconcile.Result, retErr e
...
-	// handle rollout target/source change,
-	if appRollout.Status.LastUpgradedTargetAppRevision != "" &&
-		appRollout.Status.LastUpgradedTargetAppRevision != targetAppRevisionName ||
-		(appRollout.Status.LastSourceAppRevision != "" && appRollout.Status.LastSourceAppRevision != sourceAppRevisionName) {
+	// handle rollout target/source change (only if it's not deleting already)
+	if appRollout.Status.RollingState != v1alpha1.RolloutDeletingState &&
+		((appRollout.Status.LastUpgradedTargetAppRevision != "" &&
+			appRollout.Status.LastUpgradedTargetAppRevision != targetAppRevisionName) ||
+			(appRollout.Status.LastSourceAppRevision != "" && appRollout.Status.
+				LastSourceAppRevision != sourceAppRevisionName)) {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/applicationrollout/applicationrollout_controller.go:120]: this is too complex to read, please write a function

**Review by wonderflow — COMMENTED** [review]
please consider the panic statement

File: apis/standard.oam.dev/v1alpha1/rollout_state.go (hunk 1 above)
Comment (ryanzhang-oss) [reply apis/standard.oam.dev/v1alpha1/rollout_state.go:269]: panic makes sense in terms of state transition but it is a problem when all controllers run in the same executable

**Review by wonderflow — APPROVED** [review]

**Pull request merged**